src/simplify_reshapes.cpp

```diff
@@ -949,6 +981,76 @@ struct find_reshape_reshape_dot
     }
 };
 
+struct find_reshape_const_dot
+{
+    auto matcher() const
+    {
+        return match::name("dot")(
+            match::used_once(),
+            match::either_arg(0, 1)(match::name("reshape").bind("rsp"),
+                                    match::skip_broadcasts(match::is_constant().bind("constant"))));
```

Comment:
I dont think `skip_broadcasts` is needed when using `is_constant`. However, is it necessary to check for a const? Can we just match anything?

Reply:
well, I want to bind "constant" to whatever was before the broadcast if there is a broadcast, if I dont do it I'll have to add a couple if statements in the apply method to check when the input is broadcasted vs not. 

I guess it doesnt have to be a const as long as any batch dimensions of the input are broadcasted. Otherwise, this logic for modifying the broadcast wouldnt work. Ie. for the gemm thats part of the attention, this does not work (for that scenario you would have to reshape the other input)

Reply:
I dont see a reason for it to be a constant. As long as we are expanding the batch dimension there shouldn't be a problem. The only case is that we would need to rewrite `reshape(multibroadcast(...))` if we are doing calculations to this matrix so broadcasting would be preserved.

Reply:
Ya it doesnt have to be a constant, updated.